### PR TITLE
Bug 1867848: Return empty properties and dependencies in ListBundles responses.

### PR DESCRIPTION
**Pull request by benluddy**
Bundles that don't have any properties/dependencies should return nil
or [] for those fields, but due to an error in handling NULL column
values, a single zero-valued entry was being returned.

### CODE REVIEW

File: pkg/sqlite/query.go [hunk 1]
```diff
@@ -1028,29 +1047,29 @@ func (s *SQLQuerier) ListBundles(ctx context.Context) (bundles []*api.Bundle, er
...
 			list = append(list, entry)
 		}
 	}
 	return list
 }
 
 func uniqueProps(props []*api.Property) []*api.Property {
-	keys := make(map[string]bool)
-	list := []*api.Property{}
+	keys := make(map[string]struct{})
```
Comment (kevinrizza) [inline pkg/sqlite/query.go:1067]: 👍

File: pkg/sqlite/query_test.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+	"testing"
+
+	"github.com/operator-framework/operator-registry/pkg/api"
+	"github.com/operator-framework/operator-registry/pkg/sqlite"
+	"github.com/operator-framework/operator-registry/pkg/sqlite/sqlitefakes"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListBundles(t *testing.T) {
+	const query = `SELECT DISTINCT channel_entry.entry_id, operatorbundle.bundle, operatorbundle.bundlepath,
```
Comment (kevinrizza) [inline pkg/sqlite/query_test.go:17]: Am I misreading something or is this const just not used? Is it injected into something somehow?

Comment (benluddy) [reply pkg/sqlite/query_test.go:17]: Nope, good catch. The ListBundles impl calls out to other methods on SQLQuerier, which in turn make other database queries, so I was originally going to set the stub behavior based on the query arg. I ended up stubbing based on invocation count instead and didn't use it.

File: pkg/sqlite/query.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -o sqlitefakes/fake_rowscanner.go . RowScanner
```
Comment (njhale) [inline pkg/sqlite/query.go:1]: 👍

File: pkg/sqlite/query.go [hunk 4]
```diff
@@ -918,23 +942,25 @@ func (s *SQLQuerier) ListBundles(ctx context.Context) (bundles []*api.Bundle, er
 	}
 	defer rows.Close()
 
-	bundles = []*api.Bundle{}
+	var bundles []*api.Bundle
 	bundlesMap := map[string]*api.Bundle{}
 	for rows.Next() {
-		var entryID sql.NullInt64
```
Comment (njhale) [inline pkg/sqlite/query.go:924]: 👍

File: pkg/sqlite/query_test.go [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+	"context"
+	"database/sql"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/operator-framework/operator-registry/pkg/api"
+	"github.com/operator-framework/operator-registry/pkg/sqlite"
+	"github.com/operator-framework/operator-registry/pkg/sqlite/sqlitefakes"
+	"github.com/stretchr/testify/assert"
```
Comment (njhale) [inline pkg/sqlite/query_test.go:13]: nit: import order

File: pkg/sqlite/query.go [hunk 6]
```diff
@@ -1028,29 +1047,29 @@ func (s *SQLQuerier) ListBundles(ctx context.Context) (bundles []*api.Bundle, er
...
 	}
 
-	return
+	return bundles, nil
 }
 
 func unique(deps []*api.Dependency) []*api.Dependency {
-	keys := make(map[string]bool)
-	list := []*api.Dependency{}
+	keys := make(map[string]struct{})
```
Comment (njhale) [inline pkg/sqlite/query.go:1054]: 👍

**Pull request merged**